apps/prairielearn/python/traverse_test.py

```diff
@@ -316,3 +316,11 @@ def replace(e) -> ElementReplacement:
 
     html = traverse_and_replace("<p></p>", replace)
     assert html == "<p></p>"
+
+
+def test_traverse_and_replace_script() -> None:
+    def replace(e) -> ElementReplacement:
+        return e
+
+    html = traverse_and_replace('<script>""</script>', replace)
+    assert html == '<script>""</script>'
```

Comment:
It's a good idea to have this test, but you should probably also have cases like scripts/styles that include HTML entities inside strings, e.g.,:

```html
<script>const test = "&quot;";</script>
<style>.class-a > .class-b::before { content: "a &gt; b"; }</style>
```

as well as regular content that has HTML entities:

```html
<p>This is &quot;good&quot;</p>
```

Reply:
There is already a test for the second, but I think it might be impossible for the first case you list to pass given the way the renderer is written 😰

Reply:
I just tested locally and it works as expected...